catsim/estimation.py

```diff
@@ -240,3 +241,136 @@ def estimate(
         self._evaluations = res.nfev
 
         return res.x[0]
+
+class BayesianEstimator(Estimator):
+    """Estimator that uses Bayesian estimation for ability estimation
+
+    :param prior_mean: mean of the prior distribution
+    :param prior_sd: standard deviation of the prior distribution
+    :param bounds: bounds of the ability estimate
+    :param verbose: verbosity level of the estimation
+    """
+
+    def __str__(self):
+        return 'Bayesian Estimator'
+
+    def __init__(self,
+            precision: int = 6,
+            prior_mean: float = 0.,
+            prior_sd: float = 1.,
+            bounds: tuple = (-6, 6),
+            verbose: bool = False
+            ):
+        super().__init__()
+        self._precision = precision
+        self._prior_mean = prior_mean
+        self._prior_sd = prior_sd
+        self._verbose = verbose
+        self._evaluations = 0
+        self._calls = 0
+        self._bounds = bounds
+
+    @property
+    def calls(self) -> float:
```

Comment:
Yes, true - evaluations() is called when the verbose option is set to True. In general, I thought of subclassing the HillClimbingEstimator but I'm not too familiar doing this, so that's why I refrained, and these copies exist now.